plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/options/PortLabelPlacement.java

```diff
@@ -9,19 +9,93 @@
  *******************************************************************************/
 package org.eclipse.elk.core.options;
 
+import java.util.EnumSet;
+import java.util.Set;
+
 /**
- * Definition of port label placement strategies. The chosen strategy determines whether port
- * labels are placed inside or outside of the respective node.
- *
- * @author jjc
+ * Options for controlling how port labels are placed by layout algorithms. The corresponding layout
+ * option will usually accept an {@link EnumSet} over this enumeration, theoretically allowing
+ * arbitrary and even contradictory subsets of options to be set. Note that you are restricted to
+ * the following combinations if you want your choice to make sense:
```

Comment:
I added an 'isValid'  method to the enum which is checked when retrieving the option in `NodeContext`. Any suggestion for a better/other place to check it?

While at it I also added it for `NodeLabelPlacement` as well. The change may result in graphs failing that have been working before since so far we failed silently. However, I feel it's better to let the user know the configuration is off in this case.

Reply:
This is a bit of a philosophical question. Ideally, I'd vote for failing as early as possible. What that means is up for debate, though.

At one extreme, one would add additional `IGraphElementVisitor`s to check for configuration errors. Pro: we'd fail even before a layout algorithm is run. Contra: we would either need a great deal of visitors and some sort of registry mechanism, or we'd have a couple of large visitors that combine concerns from very different parts of our code, making everything harder to maintain.

The other extreme is checking for configuration errors as soon as they become relevant, which is what you've done here. Given that this keeps concerns separated and that it's easier to maintain, I think this is the way to go.

tl;dr: well done. 😄